### PR TITLE
Update example to use dag decorator tag

### LINKED ISSUE #12269: Update TaskFlow API example to use `@dag` decorator
**Description**

Update https://github.com/apache/airflow/blob/master/airflow/example_dags/tutorial_taskflow_api_etl.py tp use `@dag` decorator instead of leveraging context manager to make it more TaskFlow-y.

### PR DESCRIPTION
**Pull request by eflann**
Update TaskFlow API example to use `@dag` decorator

Closes #12269

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/example_dags/tutorial_taskflow_api_etl.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
 # The DAG object; we'll need this to instantiate a DAG
 from airflow import DAG
```
Comment (casassg) [inline airflow/example_dags/tutorial_taskflow_api_etl.py]: This is no longer needed!
```suggestion
```

File: airflow/example_dags/tutorial_taskflow_api_etl.py [hunk 2]
```diff
@@ -45,14 +46,8 @@
...
-with DAG(
-    'tutorial_taskflow_api_etl_dag',
-    default_args=default_args,
-    description='TaskFlow API ETL DAG tutorial',
-    schedule_interval=None,
-    start_date=days_ago(2),
-    tags=['example'],
-) as dag:
+@dag(default_args=default_args, schedule_interval=None, start_date=days_ago(2))
+def tutorial_taskflow_api_etl():
```
Comment (casassg) [inline airflow/example_dags/tutorial_taskflow_api_etl.py:40]: Documentation in `@dag` decorator should  be a function `__doc__`
```suggestion
def tutorial_taskflow_api_etl():
  """### TaskFlow API Tutorial Documentation
  This is a simple ETL data pipeline example which demonstrates the use of the TaskFlow API
  using three simple tasks for Extract, Transform, and Load.
  Documentation that goes along with the Airflow TaskFlow API tutorial located
  [here](https://airflow.apache.org/docs/stable/tutorial_taskflow_api.html)
  """
```

**Review by casassg — COMMENTED** [review]
Let's make sure to update tutorial doc as well.

Also you will need to:
- Move dag documentation to function docstring
- Remove `dag.doc_md`
- Use `@task` instead of `@dag.task`
- Remove import of `DAG` as it's not longer needed

**Comment by eflann** [comment]
1. Where do you go to update the tutorial doc? 
2. Just noticed this returns a 404 for me: https://airflow.apache.org/docs/stable/tutorial_taskflow_api.html

(Really really new around here. Thanks for bearing with me.)

File: airflow/example_dags/tutorial_taskflow_api_etl.py (hunk 2 above)
Comment (kaxil) [reply airflow/example_dags/tutorial_taskflow_api_etl.py:40]: +1

**Comment by kaxil** [comment]
> 2\. https://airflow.apache.org/docs/stable/tutorial_taskflow_api.html

https://airflow.readthedocs.io/en/latest/tutorial_taskflow_api.html

File: airflow/example_dags/tutorial_taskflow_api_etl.py [hunk 3]
```diff
@@ -45,22 +35,23 @@
...
-    schedule_interval=None,
-    start_date=days_ago(2),
-    tags=['example'],
-) as dag:
-    # [END instantiate_dag]
-
+@dag(default_args=default_args, schedule_interval=None, start_date=days_ago(2))
+# [END instantiate_dag]
+def tutorial_taskflow_api_etl():
     # [START documentation]
```
Comment (ashb) [inline airflow/example_dags/tutorial_taskflow_api_etl.py:41]: We'll need to see how this renders in context -- we may need to make it clearer what this does

**Comment by ryw** [comment]
what is left to do on this one?

**Comment by casassg** [comment]
I think it probably requires to restructure the tutorial to explain using docstrings for documentation.

**Pull request closed without merging**